pkg/apiclient/workflow/workflow.proto

```diff
@@ -30,9 +30,11 @@ message WorkflowGetRequest {
 
 message WorkflowListRequest {
   string namespace = 1;
-  k8s.io.apimachinery.pkg.apis.meta.v1.ListOptions listOptions = 2;
+  string namePrefix = 2;
+  string namePattern = 3;
+  k8s.io.apimachinery.pkg.apis.meta.v1.ListOptions listOptions = 4;
   // Fields to be included or excluded in the response. e.g. "items.spec,items.status.phase", "-items.status.nodes"
-  string fields = 3;
+  string fields = 5;
```

Comment:
Changing the tag number of the existing fields might lead to some issues of decoding of current existing client, as mentioned in the [proto 3 language guide](https://protobuf.dev/programming-guides/proto3/#updating):
> “Changing” the field number is equivalent to deleting the field and adding a new field with the same type.

and here are [consequences of reusing field numbers](https://protobuf.dev/programming-guides/proto3/#consequences), so I think we might just simply add the new fields without changing the existing ones:

```
message WorkflowListRequest {
  string namespace = 1;
  k8s.io.apimachinery.pkg.apis.meta.v1.ListOptions listOptions = 2;
  string fields = 3;
  string namePrefix = 4;
  string namePattern = 5;
```